code/z_adventurezones/lavamoon.dm

```diff
@@ -1211,52 +1211,132 @@ var/global/iomoon_blowout_state = 0 //0: Hasn't occurred, 1: Moon is irradiated
 	density = 0
 	var/id = null
 	var/broken = FALSE
+	mat_appearances_to_ignore = list("negative matter")
+	mat_changename = FALSE
+	appearance_flags = KEEP_TOGETHER
+
+/obj/ladder/broken
+	name = "broken ladder"
+	desc = "it's too damaged to climb."
+	icon_state = "ladder_wall_broken"
+	broken = TRUE
+
+/obj/ladder/embed
+	name = "gap in the wall"
+	icon_state = "wall_embed"
+	desc = "A section of this wall appears to be missing. Entering it might take you somewhere."
+	plane = PLANE_WALL
+	var/hidden = FALSE
+
+/obj/ladder/embed/climb(mob/user as mob)
+	var/obj/ladder/otherLadder = src.get_other_ladder()
+	if (!istype(otherLadder))
+		boutput(user, "You try to enter the gap in the wall, but seriously fail! Perhaps there's nowhere to go?")
+		return
+	boutput(user, "You enter the gap in the wall.")
+	user.set_loc(get_turf(otherLadder))
 
-	broken
-		name = "broken ladder"
-		desc = "it's too damaged to climb."
-		icon_state = "ladder_wall_broken"
-		broken = TRUE
-
-	New()
-		..()
-		if (!id)
-			id = "generic"
-
-		src.update_id()
-
-	proc/update_id(new_id)
-		if(new_id)
-			src.id = new_id
-		src.tag = "ladder_[id][src.icon_state == "ladder" ? 0 : 1]"
-
-	proc/get_other_ladder()
-		RETURN_TYPE(/atom)
-		. = locate("ladder_[id][src.icon_state == "ladder"]")
+/obj/ladder/embed/New()
+	. = ..()
+	src.UpdateIcon()
 
-	attack_hand(mob/user)
-		if (src.broken) return
-		if (user.stat || user.getStatusDuration("weakened") || BOUNDS_DIST(user, src) > 0)
-			return
-		src.climb(user)
+/obj/ladder/embed/update_icon()
+	. = ..()
+	if (!src.hidden)
+		var/turf/T = get_step(src,NORTH)
+		if (!istype(T,/turf/simulated/wall) && !istype(T,/turf/unsimulated/wall))
+			src.alpha = 0 // if there's no wall above us, hide in a way that the turf can update us
+			src.mouse_opacity = FALSE
```

Comment:
if you want a thing to be both unclickable and invisible the easier way to do this is to set invisibility
`APPLY_ATOM_PROPERTY(src, PROP_MOB_INVISIBILITY, "hidden", INVIS_ALWAYS)`

Reply:
unfortunately INVIS_ALWAYS and INVIS_ALWAYS_ISH ended up being TOO invisible and the ladder never ended up in orange(1)

Reply:
:melterfrog: